storage/storage.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * A simple storage utility to save the user config as json file
+ * 
+ */
+const electron = require('electron');
+const path = require('path');
+const fs = require('fs');
+ class Store {
+  constructor(opts = {}) {
+    let defaultConfig = {
+      configName: opts.configName || 'user',
+      defaults: {
+        alwaysWindowOnTop: true
+      }
+    }
+     opts.defaults = Object.assign(defaultConfig.defaults, opts.defaultConfig);
+     const userDataPath = (electron.app || electron.remote.app).getPath('userData');
+    // Use the `configName` property to set the file name and path.join to bring it all together as a string
+    this.path = path.join(userDataPath, opts.configName + '.json');
+    
+    this.data = parseDataFile(this.path, opts.defaults);
+  }
+  
+  get(key) {
+    return this.data[key];
+  }
+  
+  set(key, val) {
+    this.data[key] = val;
+    try{
+      fs.writeFileSync(this.path, JSON.stringify(this.data));
+    }catch(err) {
+      console.error('Failed to save user config');
+    }
+  }
+}
+ function parseDataFile(filePath, defaults) {
+  try {
+    return JSON.parse(fs.readFileSync(filePath));
+  } catch(error) {
+    return defaults;
+  }
+}
+ module.exports = Store;
```

Comment:
I think a new object of this should be created here and then exported such that every file importing this will have one singleton object rather than multiple objects which might lead to an invalid state in the file during saving in parallel situations.

Reply:
If we do so we can create only one config file, what if i want to create another configuration for something else. I do one thing, i export store as well as default config instance
`
 export.default = new Storage({})
 export.Storage = Storage
`

Is that ok for you.

Reply:
I was thinking why not add one optional parameter in the set method which will receive the name of the file to save on.

Reply:
Then what about retrieving value in get method, Should we need to pass a file name  every time i want to get a value. If so we have to read from file for every get or we have to maintain the map of fileName against file data. Is that fine.

Reply:
Let's keep it in this way right now. We will update it later on the requirement basis.

Reply:
I mean the solution you provided for this.

Reply:
Ok, then i am resolving this conversation.